CMake minimum version

This line seems to cause issues with configuring in CMake 3.15:
https://github.com/SSAGESLabs/openmm-dlext/blob/2ec5ff935cdd8d2adfca19a0713691f862b58a33/cmake/Modules/FindOpenMM.cmake#L66

Here's the relevant error message. I confirmed it works with CMake 3.16+, so one option is to update the minimum version from 3.13 to 3.16
```
CMake Error at cmake/Modules/FindOpenMM.cmake:66 (target_include_directories):
  target_include_directories called with non-compilable target type
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)


-- Found CUDA: /software/cuda-10.1-el7-x86_64 (found version "10.1")
CMake Error at cmake/Modules/FindOpenMM.cmake:85 (target_compile_definitions):
  target_compile_definitions called with non-compilable target type
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)


CMake Error at cmake/Modules/FindOpenMM.cmake:86 (target_include_directories):
  target_include_directories called with non-compilable target type
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)


CMake Error at cmake/Modules/FindOpenMM.cmake:89 (target_include_directories):
  target_include_directories called with non-compilable target type
Call Stack (most recent call first):
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,4 @@
-cmake_minimum_required(VERSION 3.13..3.20)
-#
-#if(${CMAKE_VERSION} VERSION_LESS 3.12)
-#    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
-#endif()
+cmake_minimum_required(VERSION 3.16..3.22)
 
 # Set-up project
 project(OpenMMDLExt LANGUAGES CXX)

